smack-websocket/src/main/java/org/jivesoftware/smack/websocket/implementations/WebsocketInterface.java

```diff
@@ -0,0 +1,31 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket.implementations;
+
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.packet.TopLevelStreamElement;
+import org.jivesoftware.smack.websocket.WebsocketException;
+
+public interface WebsocketInterface {
+
+    void handshake() throws InterruptedException, SmackException, XMPPException, WebsocketException;
```

Comment:
Should be `handshake(WebsocketRemoteConnectionEndpoint endpoint)` and the for loop should be performed one layer above.